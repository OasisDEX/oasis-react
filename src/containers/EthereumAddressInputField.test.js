/* global shallow describe it expect */
/* eslint-disable import/first */
import React from 'react';
import Immutable from 'immutable';

import {
  EthereumAddressInputFieldWrapper,
  mapStateToProps,
  mapDispatchToProps
} from './EthereumAddressInputField';
import { shallow } from 'enzyme';

describe('(Container) EthereumAddressInputField', () => {
  const state = Immutable.fromJS(global.stateMock);
  const initialProps = mapStateToProps(state);
  const initialActions = mapDispatchToProps(x => x);
  const props = {
    ...initialActions,
    ...initialProps,
    fieldName: 'exampleFieldName'
  };

  it('will receive right props', () => {
    expect(initialProps).toMatchSnapshot();
  });


  it('will receive right actions', () => {
    expect(initialActions).toMatchSnapshot();
  });

  it('should render', () => {
    const wrapper = shallow(
      <EthereumAddressInputFieldWrapper {...props}/>
    );
    expect(wrapper).toMatchSnapshot();
  });

});
