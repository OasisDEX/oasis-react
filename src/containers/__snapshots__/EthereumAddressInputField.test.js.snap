// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Container) EthereumAddressInputField should render 1`] = `
<Field
  component={[Function]}
  required={true}
  validate={[Function]}
/>
`;

exports[`(Container) EthereumAddressInputField will receive right actions 1`] = `
Object {
  "actions": Object {},
}
`;

exports[`(Container) EthereumAddressInputField will receive right props 1`] = `Object {}`;
